tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,601 @@
+                {"zone": "2", "type": "window", "name": "winder", "inverse": True},
+                {"zone": "3", "type": "door"},
+            ],
+            "sensors": [
+                {"zone": "4", "type": "dht"},
+                {"zone": "5", "type": "ds18b20", "name": "temper"},
+            ],
+            "switches": [
+                {
+                    "zone": "out",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+                {"zone": "6"},
+            ],
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options=device_options,
+    )
+    entry.add_to_hass(hass)
+
+    assert (
+        await async_setup_component(
+            hass,
+            konnected.DOMAIN,
+            {konnected.DOMAIN: {konnected.CONF_ACCESS_TOKEN: "1122334455"}},
+        )
+        is True
+    )
+
+    client = await aiohttp_client(hass.http.app)
+
+    # Test updating a binary sensor
+    resp = await client.post(
+        "/api/konnected/device/112233445566",
+        headers={"Authorization": "Bearer abcdefgh"},
+        json={"zone": "1", "state": 0},
+    )
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"message": "ok"}
+    await hass.async_block_till_done()
+    assert hass.states.get("binary_sensor.konnected_445566_zone_1").state == "off"
+
+    resp = await client.post(
+        "/api/konnected/device/112233445566",
+        headers={"Authorization": "Bearer abcdefgh"},
+        json={"zone": "1", "state": 1},
+    )
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"message": "ok"}
+    await hass.async_block_till_done()
+    assert hass.states.get("binary_sensor.konnected_445566_zone_1").state == "on"
+
+    # Test updating sht sensor
+    resp = await client.post(
+        "/api/konnected/device/112233445566",
+        headers={"Authorization": "Bearer abcdefgh"},
+        json={"zone": "4", "temp": 22, "humi": 20},
+    )
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"message": "ok"}
+    await hass.async_block_till_done()
+    assert hass.states.get("sensor.konnected_445566_sensor_4_humidity").state == "20"
+    assert (
+        hass.states.get("sensor.konnected_445566_sensor_4_temperature").state == "22.0"
+    )
+
+    resp = await client.post(
+        "/api/konnected/device/112233445566",
+        headers={"Authorization": "Bearer abcdefgh"},
+        json={"zone": "4", "temp": 25, "humi": 23},
+    )
+    assert resp.status == 200
+    result = await resp.json()
+    assert result == {"message": "ok"}
+    await hass.async_block_till_done()
+    assert hass.states.get("sensor.konnected_445566_sensor_4_humidity").state == "23"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)